src/selfie_avoiding_obstacles/src/road_obstacle_detector.cpp

```diff
@@ -8,26 +8,45 @@
 Road_obstacle_detector::Road_obstacle_detector(const ros::NodeHandle &nh, const ros::NodeHandle &pnh)
     : nh_(nh)
     , pnh_(pnh)
-    , timer_duration_(0.3)
-    , is_time_calculated_for_overtake_(false)
     , received_road_markings_(false)
     , maximum_distance_to_obstacle_(0.5)
 {
   pnh_.param<bool>("visualization", visualization_, false);
   pnh_.param<float>("maximum_length_of_obstacle", maximum_length_of_obstacle_, 0.8);
   obstacles_sub_ = nh_.subscribe("/obstacles", 1, &Road_obstacle_detector::obstacle_callback, this);
-  markings_sub_ = nh_.subscribe("/road_markings", 1, &Road_obstacle_detector::road_markings_callback, this);
+  distance_sub_ = nh_.subscribe("/distance", 1, &Road_obstacle_detector::distanceCallback, this);
+  passive_mode_service_ = nh_.advertiseService("/avoiding_obst_set_passive", &Road_obstacle_detector::switchToPassive, this);
+  active_mode_service_ = nh_.advertiseService("/avoiding_obst_set_active", &Road_obstacle_detector::switchToActive, this);
   setpoint_pub_ = nh_.advertise<std_msgs::Float32>("/setpoint", 1);
+  speed_pub_ = nh_.advertise<std_msgs::Float32>("/speed", 1);
   pnh_.param<float>("point_min_x", point_min_x_, 0.3);
   pnh_.param<float>("point_max_x", point_max_x_, 1.1);
   pnh_.param<float>("point_min_y", point_min_y_, -1.3);
   pnh_.param<float>("point_max_y", point_max_y_, 1.3);
```

Comment:
Można zmienić nazwę parametru. Point na ROI (region of interest). Np `ROI_min_x`

Reply:
Dobry pomysł